course/auth.py

```diff
@@ -1005,34 +987,45 @@ def is_inst_id_locked(user):
                     is_inst_id_locked=is_inst_id_locked(request.user),
             )
             if user_form.is_valid():
-                user_form.save()
+                if user_form.has_changed():
+                    user_form.save()
+                    messages.add_message(request, messages.SUCCESS,
+                            _("Profile data updated."))
+                    request.user.refresh_from_db()
+
+                else:
+                    messages.add_message(request, messages.INFO,
+                            _("No change was made on your profile."))
 
-                messages.add_message(request, messages.INFO,
-                        _("Profile data saved."))
                 if request.GET.get("first_login"):
                     return redirect("relate-home")
                 if (request.GET.get("set_inst_id")
-                        and request.GET["referer"]):
+                        and request.GET.get("referer")):
                     return redirect(request.GET["referer"])
 
                 user_form = UserForm(
-                        instance=request.user,
-                        is_inst_id_locked=is_inst_id_locked(request.user))
+                    instance=request.user,
+                    is_inst_id_locked=is_inst_id_locked(request.user))
 
     if user_form is None:
-            user_form = UserForm(
-                    instance=request.user,
-                    is_inst_id_locked=is_inst_id_locked(request.user),
-            )
+        request.user.refresh_from_db()
+        user_form = UserForm(
+            instance=request.user,
+            is_inst_id_locked=is_inst_id_locked(request.user),
+        )
+    if is_inst_id_locked(request.user):
+        user_form.fields['institutional_id'].disabled = True
+        user_form.fields['institutional_id_confirm'].disabled = True
```

Comment:
How come this is not in the `UserForm` constructor?

Reply:
I also found the code was in a mess (composed by me). I'm rewriting this part.

Reply:
update.